lib/recipe/widgets/ingredient_form.dart

```diff
@@ -25,259 +42,354 @@ class IngredientForm extends StatelessWidget {
-                            .invalid
-                        ? 'Enter a measurement method'
-                        : null,
-                  ),
-                  onSaved: (dynamic str) {}, // investigate onSaved vs onChanged
-                  onChanged: (dynamic value) {
-                    context.read<RecipeBloc>().add(
-                        IngredientMeasurementMethodChanged(
-                            ingredient: state.recipes[recipeIndex]
-                                .ingredients[ingredientIndex],
-                            measurementMethod: value,
-                            recipe: state.recipes[recipeIndex]));
-                  },
-                  validator: (dynamic str) {},
-                  displayItemFn: (dynamic item) => Text(state
-                      .recipes[recipeIndex]
-                      .ingredients[ingredientIndex]
-                      .measurementMethod
-                      .value),
-                  findFn: (dynamic str) async => measurementMethods,
-                  filterFn: (dynamic item, str) =>
-                      item.toLowerCase().indexOf(str.toLowerCase()) >= 0,
-                  dropdownItemFn: (dynamic item, position, focused,
-                          dynamic lastSelectedItem, onTap) =>
-                      ListTile(
-                    title: Text(item),
-                    onTap: onTap,
-                  ),
+                onChanged: (value) {
+                  context.read<RecipeBloc>().add(IngredientOtherNameChanged(
+                      ingredient: state.recipes[widget.recipeIndex]
+                          .ingredients[widget.ingredientIndex],
+                      ingredientOtherName: value,
+                      recipe: state.recipes[widget.recipeIndex]));
+                },
+                textInputAction: TextInputAction.next,
+                textCapitalization: TextCapitalization.sentences,
+              ),
+            ),
+            TextFormField(
+              initialValue: state.recipes[widget.recipeIndex]
+                  .ingredients[widget.ingredientIndex].description,
+              decoration: InputDecoration(
+                icon: const Icon(Icons.description_rounded),
+                labelText: 'Ingredient description',
+                helperText: 'Ingredient description e.g. Big, dry, ripe etc.',
+                errorText: (state
+                                .recipes[widget.recipeIndex]
+                                .ingredients[widget.ingredientIndex]
+                                .description !=
+                            null &&
+                        state
+                                .recipes[widget.recipeIndex]
+                                .ingredients[widget.ingredientIndex]
+                                .description ==
+                            '')
+                    ? 'Enter an ingredient description e.g. Ripe'
+                    : null,
+              ),
```

Comment:
I think this bit that checks whether the field has been modified and is still empty should be refactored into a method of the Recipe to make it more readable, maybe something like
```dart
errorText: Recipe.isModifiedAndEmpty(state
                                .recipes[widget.recipeIndex]
                                .ingredients[widget.ingredientIndex]
                                .description)
                    ? 'Enter an ingredient description e.g. Ripe'
                    : null,
```

basically just have the function that checks for notNull and Empty somewhere in the backend and named expressively.